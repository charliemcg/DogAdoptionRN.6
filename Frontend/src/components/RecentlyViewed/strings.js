export default {
  recentlyViewed: "Recently Viewed"
};
