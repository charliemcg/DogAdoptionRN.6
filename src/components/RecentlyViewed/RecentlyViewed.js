import React, { Component } from "react";
import { View, Text, FlatList } from "react-native";
import { connect } from "react-redux";
import styles from "./styles";
import MiniListItem from "../MiniListItem";
import strings from "./strings";

class RecentlyViewed extends Component {
  render() {
    return (
      <View>
        <Text style={styles.recents}>{strings.recentlyViewed}</Text>
        <FlatList
          data={this.props.recentlyViewed}
          horizontal={true}
          renderItem={({ item }) =>
            // don't show the currently selected dog in the recents
            item !== this.props.selectedDog && (
              <MiniListItem
                item={item}
                navigation={this.props.navigation}
                showFav={false}
              />
            )
          }
        />
      </View>
    );
  }
}

const mapStateToProps = state => {
  return {
    recentlyViewed: state.recentlyViewed,
    selectedDog: state.selectedDog
  };
};

export default connect(
  mapStateToProps,
  null
)(RecentlyViewed);
