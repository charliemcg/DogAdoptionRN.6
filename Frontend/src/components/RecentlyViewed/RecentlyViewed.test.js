import "react-native";
import React from "react";
import RecentlyViewed from "./RecentlyViewed";
import renderer from "react-test-renderer";
import store from "../../store";

test("RecentlyViewed snapShot", () => {
  const snap = renderer.create(<RecentlyViewed store={store} />).toJSON();

  expect(snap).toMatchSnapshot();
});
