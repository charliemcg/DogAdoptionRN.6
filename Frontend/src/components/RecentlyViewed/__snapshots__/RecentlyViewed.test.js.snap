// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecentlyViewed snapShot 1`] = `
<View>
  <Text
    style={
      Object {
        "color": "#666",
        "fontSize": 22,
        "marginLeft": "1%",
      }
    }
  >
    Recently Viewed
  </Text>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;
